syntax = "proto3";
package instructions;

// Instructions (Generated by Grader)
message RegisterAssignmentInstruction {
    string public_uuid = 1;
    string course_public_uuid = 2;
    string assignment_version_uuid = 3;
}

message RegisterCriterionInstruction {
    string public_uuid = 1;
    string course_public_uuid = 2;
    string assignment_version_uuid = 3;
    string criterion_uuid = 4;
}

message AssignmentInstruction {
    string public_uuid = 1;
    string version_uuid = 2;
    string content = 3;
    uint64 version_timestamp = 4;
}

message CriterionInstruction {
    string public_uuid = 1;
    string version_uuid = 2;
    string content = 3;
    uint64 version_timestamp = 4;
}

message InvalidateInstruction {
    string instruction_version_uuid = 1;
    optional bool is_assignment_instruction = 2;
}

message InstructionInfo {
    repeated string assignment_instruction_version_uuids = 1;
    repeated string criterion_instruction_version_uuids = 2;
}

// use instruction versions
// when an instruction is invaluuidated or changed regrade all the current grades that reference it
// mark which ones we need to regrade?

// look into the past and mark grades for regrade
// new instruction
// changed instruction
// TODO: regrade all current grades using incomplete, invaluuid, or changed instructions
// submit new grade with same value if no change was needed
// invaluuidated instruction